Development/src/pages/devices/EditMatrixNamesField.js

```diff
@@ -0,0 +1,315 @@
+import React, { Fragment, useEffect, useRef, useState } from 'react';
+import { Button, IconButton, TextField, Tooltip } from '@material-ui/core';
+import get from 'lodash/get';
+import has from 'lodash/has';
+import CreateIcon from '@material-ui/icons/Create';
+import DeleteIcon from '@material-ui/icons/Delete';
+import ClearIcon from '@material-ui/icons/Clear';
+import DoneIcon from '@material-ui/icons/Done';
+import { isEmpty, set } from 'lodash';
+
+export const EditableIONameField = ({
+    defaultValue,
+    source,
+    label,
+    personalNames,
+    setPersonalNames,
+    autoFocus,
+    getLabelByLenght,
+    title,
+    ioKey,
+    ...props
+}) => {
+    const [displayRenameField, setDisplayRenameField] = useState(false);
```

Comment:
maybe change `displayRenameField, setDisplayRenameField` to `displayPersonalField, setPersonalField`